jgrapht-io/src/main/java/org/jgrapht/nio/tsplib/TSPLIBImporter.java

```diff
@@ -984,12 +984,20 @@ int compute2DGeographicalDistance(Node n1, Node n2)
     static final double PI = 3.141592; // constants according to TSPLIB95
     static final double RRR = 6378.388; // constants according to TSPLIB95
 
-    private static double computeRadiansAngle(double x)
+    static double computeRadiansAngle(double x)
     { // computation according to TSPLIB95 chapter 2.4 - Geographical distance
       // First computes decimal angle from degrees and minutes, then converts it into radian
-        double deg = Math.round(x);
-        double min = x - deg;
-        return PI * (deg + 5.0 * min / 3.0) / 180.0;
+      // See also the question regarding type GEO in TSPLIB - FAQ:
+      // http://comopt.ifi.uni-heidelberg.de/software/TSPLIB95/TSPFAQ.html
+
+        double deg = (int) x; // integer degree part
+        double min = x - deg; // decimal part represents the minutes of arc
+        if (0.60 < Math.abs(min)) {
+            throw new IllegalArgumentException(
```

Comment:
This let the import of ali535.tsp fail because it contains the following three lines:
```
187  33.56  45.80
429  48.31 -24.80
501  40.62  13.11
```
In each line one coordinate has a decimal part greater 0.6. But since the decimal part represents the minutes of arc of the degree value it must not be greater 0.6. So from my perspective the TSP-instance is malformed.

Alternatively only a corresponding message could be printed or it could configurable if the algorithm should only print an error or should throw an exception. That's not nice but at least the import of ali535 would work.
Another way would be to contact the Uni Heidelberg ask them to fix this instance.